sentinelsat/sentinel.py

```diff
@@ -407,15 +407,15 @@ def get_product_odata(self, id, full=False):
         https://github.com/SentinelDataHub/DataHubSystem/blob/master/addon/sentinel-2/src/main/resources/META-INF/sentinel-2.owl
         https://github.com/SentinelDataHub/DataHubSystem/blob/master/addon/sentinel-3/src/main/resources/META-INF/sentinel-3.owl
         """
-        url = urljoin(self.api_url, u"odata/v1/Products('{}')?$format=json".format(id))
+        url = urljoin(self.api_url, u"odata/v1/Products('{}')?$format=json".format(product_id))
         if full:
             url += '&$expand=Attributes'
         response = self.session.get(url, auth=self.session.auth)
         _check_scihub_response(response)
         values = _parse_odata_response(response.json()['d'])
         return values
 
-    def download(self, id, directory_path='.', checksum=True):
+    def download(self, product_id, directory_path='.', checksum=True):
```

Comment:
`id()` is probably one of the least used builtins and shadowing it in code is not really a problem, imo. The only use for `id()` I can imagine is for super-optimized data structure or algorithm implementations, anywhere else you should be using `x is y` anyway for reference equality checks.
`product_id` is a slightly more descriptive name, I agree. I somewhat dislike modifying the public method signature, but I would not expect anyone to be using the parameter via a keyword anyway, so it should not be an issue.

Reply:
I agree that changing public interface is not the wisest move, so it should be in the changelog. In my opinion, overshadowing any built-ins can lead to the shot in the leg,  and that's why I proposed this change in the first place.

Reply:
I feel the same about `id`, @valgur. But I think this is fine to change to get rid of any builtin-shadowing.

Reply:
I would prefer to rename it to `uuid` as that is the name DHuS refers to in its [documentation](https://scihub.copernicus.eu/twiki/do/view/SciHubUserGuide/ODataAPI#Download_full_product_knowing_it).